apps/visualization/utils.py

```diff
@@ -556,3 +564,79 @@ async def get_global_combined_indicators(filters):
     if filters:
         qs = filter_apply(filters, qs)
     return await process_combined_indicators(qs, type=GLOBAL_LEVEL)
+
+
+@sync_to_async
+def get_indicator_stats_latest(
+    emergency,
+    region,
+    indicator_id,
+    is_top=False,
+):
+    from .types import IndicatorLatestStatsType
+
+    existing_iso3 = Countries.objects.values_list('iso3', flat=True)
+
+    def get_unique_countries_data(qs):
+        data_country_map = {}
+        for item in qs:
+            if data_country_map:
+                if data_country_map.get(item['iso3'], None):
+                    pass
+                else:
+                    data_country_map[item['iso3']] = item
+            else:
+                data_country_map[item['iso3']] = item
+
+        return [
+            IndicatorLatestStatsType(
+                iso3=iso3,
+                indicator_value=map_data['indicator_value'],
+                format=map_data['format'],
+                country_name=map_data['country_name'],
+            ) for iso3, map_data in data_country_map.items()
+        ]
+
+    if indicator_id:
+        all_filters = {
+            'region': region,
+            'indicator_id': indicator_id,
+            'emergency': emergency,
+        }
+        filters = {k: v for k, v in all_filters.items() if v is not None}
+        qs = DataCountryLevelMostRecent.objects.filter(
+            **filters,
+            iso3__in=existing_iso3,
+            category='Global',
+            indicator_value__gt=0
+        ).values('iso3').annotate(
+            indicator_value=F('indicator_value'),
+            max_indicator_month=Max('indicator_month'),
+            format=F('format'),
+            country_name=F('country_name'),
+        ).order_by('-max_indicator_month', 'subvariable')
+
+    else:
+        all_filters = {
+            'emergency': emergency,
+        }
+        filters = {k: v for k, v in all_filters.items() if v is not None}
+        qs = CountryEmergencyProfile.objects.filter(
+            **filters,
+            iso3__in=existing_iso3,
+            context_indicator_value__gt=0
+        ).values('iso3').annotate(
+            indicator_value=F('context_indicator_value'),
+            max_indicator_month=Max('context_date'),
+            format=F('format'),
+            country_name=Subquery(
+                Countries.objects.filter(
+                    iso3=OuterRef('iso3')
+                ).order_by().values('country_name')[:1]
+            ),
+        ).order_by('max_indicator_month')
+    if is_top:
+        qs = qs.order_by('-indicator_value')[:5]
+    else:
+        qs = qs.order_by('indicator_value')[:5]
+    return get_unique_countries_data(qs)
```

Comment:
Let's add some documentation here why we are using different tables as well.

```suggestion
@sync_to_async
def get_indicator_stats_latest(
    emergency,
    region,
    indicator_id,
    is_top=False,
):
    from .types import IndicatorLatestStatsType

    existing_iso3 = Countries.objects.values('iso3')

    def _clean_filters(filters):
        # Filter out None values
        return {
            k: v
            for k, v in filters.items()
            if v is not None
        }

    filters = _clean_filters({
        'region': region,
        'emergency': emergency,
    })
    if indicator_id:
        filters.update({
            'indicator_id': indicator_id,
        })
        qs = DataCountryLevelMostRecent.objects.filter(
            **filters,
            iso3__in=existing_iso3,
            category='Global',
            indicator_value__gt=0
        ).order_by().values('iso3').annotate(
            indicator_value=F('indicator_value'),
            max_indicator_month=Max('indicator_month'),
            format=F('format'),
            country_name=F('country_name'),
        ).order_by('-max_indicator_month', 'subvariable')

    else:
        qs = CountryEmergencyProfile.objects.filter(
            **filters,
            iso3__in=existing_iso3,
            context_indicator_value__gt=0
        ).order_by().values('iso3').annotate(
            indicator_value=F('context_indicator_value'),
            max_indicator_month=Max('context_date'),
            format=F('format'),
            country_name=Subquery(
                Countries.objects.filter(
                    iso3=OuterRef('iso3')
                ).order_by().values('country_name')[:1]
            ),
        ).order_by('max_indicator_month')

    if is_top:
        qs = qs.order_by('-indicator_value')[:5]
    else:
        qs = qs.order_by('indicator_value')[:5]

    return [
        IndicatorLatestStatsType(
            iso3=data['iso3'],
            indicator_value=data['indicator_value'],
            format=data['format'],
            country_name=data['country_name'],
        ) for data in qs.distinct('iso3')
    ]
```

Reply:
`return [
        IndicatorLatestStatsType(
            iso3=data['iso3'],
            indicator_value=data['indicator_value'],
            format=data['format'],
            country_name=data['country_name'],
        ) for data in qs.distinct('iso3')
    ]`
@thenav56 this codes gives an error  "Cannot create distinct fields once a slice has been taken.". So I tried 
`if is_top:
        qs = qs.distinct('iso3').order_by('-indicator_value')[:5]
    else:
        qs = qs.distinct('iso3').order_by('indicator_value')[:5]`
This leads to error '"annotate() + distinct(fields) is not implemented."
So, may be we should use the previous logic.

Reply:
We don't need the `.distinct('iso3')` as we are using **iso3** for **GROUP BY**. Let's remove that and it should work.

Reply:
@thenav56 Sir, Done !